aodncore/pipeline/steps/harvest.py

```diff
@@ -67,75 +66,55 @@ def run(self, pipeline_files):
         addition_slices = additions.get_slices(self.slice_size)
 
         for file_slice in deletion_slices:
-            try:
-                deletion_map = self.match_harvester_to_files(file_slice)
-                if not deletion_map:
-                    self._logger.info('No files to delete, proceeding to add matching files')
-                else:
-                    self.run_deletions(deletion_map, self.tmp_base_dir)
-            except SystemCommandFailedError as e:
-                self._logger.error(
-                    "File harvest deletion failed with exception: {e}".format(e=format_exception(e)))
-                # exit out of processing the current slice - is this the right behaviour on deletion error?
-                raise
+            deletion_map = self.match_harvester_to_files(file_slice)
+            self.validate_harvester_mapping(file_slice, deletion_map)
+            self.run_deletions(deletion_map, self.tmp_base_dir)
 
         for file_slice in addition_slices:
-            try:
-                addition_map = self.match_harvester_to_files(file_slice)
-                self.validate_file_handling(file_slice, addition_map)
-
-                for matched_files in addition_map.values():
-                    for f in matched_files:
-                        self._logger.info("Adding file with destination path: {}".format(f.dest_path))
-
-                self.run_additions(addition_map, self.tmp_base_dir)
-            except SystemCommandFailedError as e:
-                # exit out of processing the current slice and stop the talend-trigger
-                self._logger.error(
-                    "File harvest addition failed with exception: {e}".format(e=format_exception(e)))
-                raise
+            addition_map = self.match_harvester_to_files(file_slice)
+            self.validate_harvester_mapping(file_slice, addition_map)
+            self.run_additions(addition_map, self.tmp_base_dir)
 
     def match_harvester_to_files(self, file_list):
         harvester_map = OrderedDict()
 
         for harvester, config_item in self._config.trigger_config.items():
             matched_files = PipelineFileCollection()
-            extra_params = []
 
             for event in config_item['events']:
                 for config_type, patterns in event.items():
                     if config_type == 'regex':
                         for pattern in patterns:
                             matched_files_for_pattern = file_list.filter_by_attribute_regex('dest_path', pattern)
+                            match_list = [{'file': f.src_path, 'deletion': f.is_deletion} for f in
+                                          matched_files_for_pattern]
+                            self._logger.sysinfo(
+                                "harvester '{harvester}' matched files: {match_list}".format(harvester=harvester,
+                                                                                             match_list=match_list))
```

Comment:
We need to skip this logging when `match_list` is empty, otherwise it adds a pointless log line for every harvester * every regex.